### PR TITLE
[docs] - Update Asset Checks overview page (DOC-279)

### PR DESCRIPTION
**Pull request by erinkcochran87**
## Summary & Motivation

This PR adds some detail to the Asset Checks concept page, as per Jarred's request in [DOC-279](https://linear.app/dagster-labs/issue/DOC-279/[docs]-[asset-checks]-update-asset-check-overview-page).

### CODE REVIEW

File: docs/content/concepts/assets/asset-checks.mdx [hunk 1]
```diff
@@ -1,44 +1,107 @@
...
+Before continuing, you should be familiar with:
+
+- [Assets](/concepts/assets/software-defined-assets)
 
 ---
 
 ## How it works
 
-Asset checks are [defined in code](/concepts/assets/asset-checks/define-execute-asset-checks) alongside the asset(s) they target. You can use Python to define them, or if using dbt, load your [dbt tests into Dagster as asset checks](/integrations/dbt/reference#loading-dbt-tests-as-asset-checks). Refer to the [Defining and executing asset checks](/concepts/assets/asset-checks/define-execute-asset-checks) guide for more info.
+An asset check is a data quality check, [defined in code](/concepts/assets/asset-checks/define-execute-asset-checks), that targets one or more Dagster assets. You can use Python to define them, or if using dbt, load your [dbt tests into Dagster as asset checks](/integrations/dbt/reference#loading-dbt-tests-as-asset-checks).
```
Comment (sryza) [inline docs/content/concepts/assets/asset-checks.mdx:50]: While most asset checks are data quality checks, some are not. E.g. freshness checks don't really qualify as data quality checks. Someone could also write an asset check to detect private data making it to places that it's not supposed to.

**Review by sryza — COMMENTED** [review]
One nitpicky comment. Otherwise looks good to go!

File: docs/content/concepts/assets/asset-checks.mdx (hunk 1 above)
Comment (erinkcochran87) [reply docs/content/concepts/assets/asset-checks.mdx:50]: That's fair! I'll update this to be a bit more broad.

**Review by sryza — APPROVED** [review]
Nice

**Pull request merged**